extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/VarBinaryChunkInputStreamGenerator.java

```diff
@@ -78,30 +228,27 @@ public void close() {
             if (chunk instanceof PoolableChunk) {
                 ((PoolableChunk) chunk).close();
             }
-            if (offsets != null) {
-                offsets.close();
+            if (byteStorage != null) {
+                byteStorage.offsets.close();
             }
         }
     }
 
     @Override
     public DrainableColumn getInputStream(final StreamReaderOptions options, final @Nullable RowSet subset) throws IOException {
         computePayload();
-        return new ObjectChunkInputStream(options, offsets, bytes, subset);
+        return new ObjectChunkInputStream(options, byteStorage, subset);
     }
 
     private class ObjectChunkInputStream extends BaseChunkInputStream {
         private int cachedSize = -1;
-        private final byte[] myBytes;
-        private final IntChunk<ChunkPositions> myOffsets;
+        private final ByteStorage myByteStorage;
 
         private ObjectChunkInputStream(
                 final StreamReaderOptions options,
-                final IntChunk<ChunkPositions> myOffsets,
-                final byte[] myBytes, final RowSet subset) {
+                final ByteStorage myByteStorage, final RowSet subset) {
             super(chunk, options, subset);
-            this.myBytes = myBytes;
-            this.myOffsets = myOffsets;
+            this.myByteStorage = myByteStorage;
```

Comment:
This is a non-static inner class -- so we have access to the outer `byteStorage`. I don't know why I did that the first time around. Maybe at one point it was a static inner class...?